### PR TITLE
Endpoint updates

### PR DESCRIPTION
**Pull request by DakCrowder**
- Adds a number of new parameters to the flightctl_info module that can be passed to GET endpoints
- Modifies the flightctl_info response type to include additional info beyond the list of response data
- Adds an enrollment config info module
  - The config isn't a 'resource' in the same way that devices, fleets etc and the response is fairly unique/specific to only fetching the one resource so I broke it out into a separate small module similar to the certificates.

NOTE: This will need to be refactored if https://github.com/flightctl/flightctl-ansible/pull/11 is merged

### CODE REVIEW

File: plugins/module_utils/options.py [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        if self.rendered and self.kind is not Kind.DEVICE:
+            raise ValidationException("Rendered field is only valid for Device kind")
+        if self.fleet_name and self.kind is not Kind.TEMPLATE_VERSION:
+            raise ValidationException("Fleet name field is only valid for TemplateVersion kind")
+        if self.summary_only:
+            if self.kind is not Kind.DEVICE:
+                raise ValidationException("Summary Only field is only valid for Device kind")
+            if self.name:
+                raise ValidationException("Summary Only field is not valid when fetching one Device")
+        if self.summary:
```
Comment (DakCrowder) [inline plugins/module_utils/options.py:72]: I don't love how this looks w/ so many fields and validations.  This pr is getting quite large but I will probably try to come back and refactor, possibly consolidating some of the logic at the field level or offload this to the ansible argument spec validator https://docs.ansible.com/ansible/latest/reference_appendices/module_utils.html#argumentspecvalidator

File: plugins/modules/flightctl_info.py [hunk 2]
```diff
@@ -25,9 +25,47 @@
     description:
       - Use to specify an object name.
     type: str
+  fleet_name:
+    description:
+      - Use to specify a fleet name that owns the associated resources. Only applicable when kind is TemplateVersions.
```
Comment (alinabuzachis) [inline plugins/modules/flightctl_info.py:30]:
```suggestion
      - Use to specify a fleet name that owns the associated resources. Only applicable when O(kind=TemplateVersions).
```

File: plugins/modules/flightctl_info.py [hunk 3]
```diff
@@ -25,9 +25,47 @@
...
+    type: str
   label_selector:
     description:
-      - A selector to restrict the list of returned objects by their labels.
+      - A selector to restrict the list of returned objects by their labels.  Accepts a a comma-separated list of key1=value1,key2=value2
+    type: str
+  field_selector:
+    description:
+      - A selector to filter based on object fields.  Accepts a a comma-separated list of key1=value1,key2=value2
```
Comment (alinabuzachis) [inline plugins/modules/flightctl_info.py:38]:
```suggestion
      - A selector to filter based on object fields.  Accepts a a comma-separated list of V(key1=value1,key2=value2)
```

File: plugins/modules/flightctl_info.py [hunk 4]
```diff
@@ -25,9 +25,47 @@
...
+    type: str
+  owner:
+    description:
+      - Filter results by owner, specified in kind/name format.
+    type: str
+  rendered:
+    description:
+      - Return the rendered device configuration that is presented to the device.  Only applicable when kind is Device.
```
Comment (alinabuzachis) [inline plugins/modules/flightctl_info.py:46]:
```suggestion
      - Return the rendered device configuration that is presented to the device.  Only applicable when O(kind=Device).
```

File: plugins/modules/flightctl_info.py [hunk 5]
```diff
@@ -25,9 +25,47 @@
...
+    type: bool
+  summary:
+    description:
+      - Return aggregate summary info for devices within a fleet.  Only applicable when kind is Fleet.
```
Comment (alinabuzachis) [inline plugins/modules/flightctl_info.py:50]:
```suggestion
      - Return aggregate summary info for devices within a fleet.  Only applicable when O(kind=Fleet).
```

File: plugins/modules/flightctl_info.py [hunk 6]
```diff
@@ -25,9 +25,47 @@
...
+    type: bool
+  summary_only:
+    description:
+      - Return only the summary info for devices.  Only the 'owner' and 'label_selector' parameters are supported. Only applicable when kind is Device.
```
Comment (alinabuzachis) [inline plugins/modules/flightctl_info.py:54]:
```suggestion
      - Return only the summary info for devices.  Only the O(owner) and O(label_selector) parameters are supported. Only applicable when O(kind=Device).
```

File: plugins/modules/flightctl_info.py [hunk 7]
```diff
@@ -25,9 +25,47 @@
...
+    type: bool
+  status_filter:
+    description:
+      - A filter to restrict the list of devices by the value of the filtered status key.  Only applicable when kind is Device.
```
Comment (alinabuzachis) [inline plugins/modules/flightctl_info.py:58]:
```suggestion
      - A filter to restrict the list of devices by the value of the filtered status key.  Only applicable when O(kind=Device).
```

File: plugins/modules/flightctl_info.py [hunk 8]
```diff
@@ -54,43 +92,90 @@
...
 
 
 RETURN = r"""
 result:
   description:
-    - The object(s) that exists
-  returned: success
-  type: list
-  elements: dict
+    - The list response containing the object(s) that exist and relevant metadata
```
Comment (alinabuzachis) [inline plugins/modules/flightctl_info.py:122]:
```suggestion
    - The list response containing the object(s) that exist and relevant metadata.
```

File: plugins/modules/flightctl_info.py [hunk 9]
```diff
@@ -54,43 +92,90 @@
...
+  returned: sucess
+  type: complex
   contains:
-    apiVersion:
-      description: The versioned schema of this representation of an object.
+    data:
+      description:
+        - The object(s) that exists
```
Comment (alinabuzachis) [inline plugins/modules/flightctl_info.py:128]:
```suggestion
        - The object(s) that exists.
```

File: plugins/modules/flightctl_info.py [hunk 10]
```diff
@@ -54,43 +92,90 @@
...
+          returned: When the total number of items queried is greater than C(limit) or the default limit.
+          type: str
+        remainingItemCount:
+          description: The number of subsequent items in the list which are not included in this list response.
+          returned: When the total number of items queried is greater than C(limit) or the default limit.
+          type: int
+    summary:
+      description:
+        - A summary rollup of queried objects
+      returned: When C(summary_only) is true
```
Comment (alinabuzachis) [inline plugins/modules/flightctl_info.py:170]:
```suggestion
      returned: When O(summary_only=true)
```

**Review by alinabuzachis — APPROVED** [review]

**Pull request merged**